### PR TITLE
Minc brainbrowser

### PR DESCRIPTION
**Pull request by gluneau**
- New BrainBrowser v2.5.2
- You can reset by pressing 'r'
- Load minc directly, no more conversion to raw.

### CODE REVIEW

File: modules/brainbrowser/ajax/minc.php [hunk 1]
```diff
@@ -30,7 +30,9 @@
 $header      = $_REQUEST['minc_headers'];
 $header_data = $_REQUEST['raw_data'];
 if ($header_data) {
-    passthru("minctoraw -double -normalize $minc_file");
+    $fp = fopen($minc_file, 'r');
+    fpassthru($fp);
```
Comment (driusan) [inline modules/brainbrowser/ajax/minc.php]: The `fpassthru` documentation recommends using `readfile` in situations like this:

```
If you just want to dump the contents of a file to the output buffer, without first modifying it or seeking to a particular offset, you may want to use the readfile(), which saves you the fopen() call.
```

http://php.net/manual/en/function.fpassthru.php

Comment (johnsaigle) [inline modules/brainbrowser/ajax/minc.php]: I also like `readfile`!

Comment (gluneau) [inline modules/brainbrowser/ajax/minc.php]: Ok, I'll make it readfile.  

Here are 8 other instances of fpassthru:
https://github.com/aces/Loris/search?utf8=%E2%9C%93&q=fpassthru%28%24fp%29&type=

Comment (driusan) [inline modules/brainbrowser/ajax/minc.php]: you can just make a ticket for those, it's a minor issue

**Comment by MounaSafiHarab** [comment]
@gluneau 

works really well. Found no issues. So passed manual test.
But here are a couple of comments:
1) since 'r' resets, why do we still need the reset button then? (maybe there is value in having both)
2) It is taking a longggg time to load scans. Aren't we then back to the initial problem we had, and which we thought this upgrade will solve for us?

I will bring this up in the Imaging meeting on Tuesday.

**Comment by gluneau** [comment]
Our assomption was that the increased loading time was caused by file size that were 8 times bigger on average when using the "double" option.

```
-rw-rw-r-- 1 lorisadmin lorisadmin 128K Jun  9 08:51 ibis_999434_V03_dti_001_byte.raw
-rw-rw-r-- 1 lorisadmin lorisadmin 1.0M Jun  9 08:51 ibis_999434_V03_dti_001_double.raw
-rw-rw---- 1 lorisadmin www-data   392K Jun 13  2016 ibis_999434_V03_dti_001.mnc
-rw-rw---- 1 lorisadmin www-data   513K Jun 13  2016 ibis_999434_V03_dti_001.nii
-rw-rw-r-- 1 lorisadmin lorisadmin 8.8M Jun  9 08:47 ibis_999434_V03_t1w_001_byte.raw
-rw-rw-r-- 1 lorisadmin lorisadmin  70M Jun  9 08:44 ibis_999434_V03_t1w_001_double.raw
-rw-rw---- 1 lorisadmin www-data   8.4M Jun 13  2016 ibis_999434_V03_t1w_001.mnc
-rw-rw---- 1 lorisadmin www-data    36M Jun 13  2016 ibis_999434_V03_t1w_001.nii
-rw-rw-r-- 1 lorisadmin lorisadmin  10M Jun  9 08:49 ibis_999434_V03_t2w_001_byte.raw
-rw-rw-r-- 1 lorisadmin lorisadmin  80M Jun  9 08:50 ibis_999434_V03_t2w_001_double.raw
-rw-rw---- 1 lorisadmin www-data   5.9M Jun 13  2016 ibis_999434_V03_t2w_001.mnc
-rw-rw---- 1 lorisadmin www-data    41M Jun 13  2016 ibis_999434_V03_t2w_001.nii
```

**Comment by driusan** [comment]
@MounaSafiHarab there's value in both a graphical and keyboard based way to do something.

But how long is "longggg"? Is it because it's taking longer to render, or is it taking the longer to download?

**Comment by gluneau** [comment]
Minc are slightly longer than byte, but have all the information the double did, but is much faster.

```
      | Byte       | Minc       | Double
------|------------|------------|-----------
 T1   | 9.5M  2.6s | 17.6M 3.9s | 24.8M 7.3s
 T2   | 9.9M  2.9s | 15.5M 3.4s | 20.1M 7.1s
 Bold | 1.6M  1.3s |  2.9M 1.6s |  5.6M 2.3s
 DTI  | 6.2M  4.4s | 10.3M 5.5s | 27.0M 9.0s
```

**Comment by driusan** [comment]
Those numbers for the size of the MINC file in MB don't make sense to me. I just checked a few random T1s on IBIS (which I assume is your data set), and they're in the 9MB range, not the 18MB range.

**Comment by gluneau** [comment]
These are MB indeed.  They are the size reported by the browser.

image

I've also included the minc_id numbers for reproducibility of my research.

**Comment by thomasbeaudry** [comment]
I have mincs that are 15MB, and have had 30MB in the past

Tom

**Comment by driusan** [comment]
@thomasbeaudry yes, minc files *can* be that size (especially for 4d datasets), but my point is that in the dataset being tested with the MINC files should be approximately the same size as the raw byte array.

@gluneau The screenshot that you posted shows 5.9MB for the (presumably, it's cut off..) MINC file, not the 17.6MB that your table does.

**Comment by gluneau** [comment]
@driusan The screenshot actually related to the DTI in byte mode, 4.4 seconds and 6.2 MB. (See the bottom line of the picture)

The metrics I gathered refer to the weight of the whole page in MB and the whole page load time in seconds.

**Comment by MounaSafiHarab** [comment]
When I loaded files the second day, they loaded much faster this time, so I am no longer sure about my initial comment that it is taking long to load.
@cmadjar, since you do QC for PreventAD, do you mind checking out this branch and seeing if files load in a reasonable amount of time? based on @gluneau's numbers, I think the benefits of having the new browser outweigh the extra second or two is takes to load the file.

**Pull request merged**